.github/workflows/linting.yml

```diff
@@ -68,6 +68,97 @@ jobs:
-        uses: actions/checkout@v4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
       - name: Check DAOS ftest tags.
         run: \[ ! -x src/tests/ftest/tags.py \] || ./src/tests/ftest/tags.py lint
+
+  flake8-lint:
+    runs-on: ubuntu-22.04
+    name: Flake8 check
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set up Python environment
+        uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d  # v5.1.0
+        with:
+          python-version: '3'
+      - name: Add parser
+        run: echo -n "::add-matcher::ci/daos-flake-matcher.json"
+      - name: Add whitespace parser
+        run: echo -n "::add-matcher::ci/daos-flakew-matcher.json"
+      - name: Add error parser
+        run: echo -n "::add-matcher::ci/daos-flakee-matcher.json"
+      - name: flake8 Lint
+        uses: py-actions/flake8@84ec6726560b6d5bd68f2a5bed83d62b52bb50ba  # v2.3.0
+        with:
+          # W503 and W504 are related as they conflict.  W503 is the preferred style and all code
+          # should be using it now.
+          ignore: 'W503'
+          exclude: 'src/control/vendor,src/client/pydaos/raw'
+          max-line-length: '100'
+      - name: flake8 Lint on SCons files.
+        uses: py-actions/flake8@84ec6726560b6d5bd68f2a5bed83d62b52bb50ba  # v2.3.0
+        with:
+          ignore: 'F821,W503,F841'
+          max-line-length: '100'
+          args: '--filename */SConscript, SConstruct'
+
+  Doxygen:
+    name: Doxygen
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install doxygen
+        run: sudo apt-get install doxygen
+      - name: Add parser
+        run: echo -n "::add-matcher::ci/daos-doxygen-matcher.json"
+      - name: Run check
+        run: doxygen Doxyfile
+      - name: 'Upload Artifact'
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3  # v4.3.1
+        with:
+          name: API Documentation
+          path: docs/doxygen/html/
+          retention-days: 1
+
+  pylint:
+    name: Pylint check
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d  # v5.1.0
+        with:
+          python-version: '3.11'
+      - name: Install core python packages
+        run: python3 -m pip install --requirement requirements.txt
+      - name: Install extra python packages
+        run: python3 -m pip install --requirement utils/cq/requirements.txt
```

Comment:
Not a regression but `requirements.txt` pulls in `utils/cq/requirements.txt` now anyway so this run step isn't needed and could be removed.

Reply:
Done